dataflow/src/domain/mod.rs

```diff
@@ -1566,6 +1574,80 @@ impl Domain {
         }
     }
 
+    fn handle_recovery(&mut self) {
+        let node_info: Vec<_> = self.nodes
+            .iter()
+            .map(|(index, node)| {
+                let n = node.borrow();
+                (index.clone(), n.global_addr(), n.is_transactional())
+            })
+            .collect();
```

Comment:
Why collect if you're directly iterating over it below anyway?

Reply:
Lifetime issues: `self.handle()` can't by called while `self.nodes` is borrowed